src/proto/robot_status_msg.proto

```diff
@@ -67,70 +75,26 @@ enum ErrorCode
     NO_ERROR = 0;
 
     // Level triggered events
-    LOW_CAP                         = 1;
-    LOW_BATTERY                     = 2;
-    HIGH_BOARD_TEMP                 = 3;
-    CHARGE_TIMEOUT                  = 4;
-    WHEEL_0_MOTOR_HOT               = 5;
-    WHEEL_1_MOTOR_HOT               = 6;
-    WHEEL_2_MOTOR_HOT               = 7;
-    WHEEL_3_MOTOR_HOT               = 8;
-    DRIBBLER_MOTOR_HOT              = 9;
-    WHEEL_0_ENCODER_NOT_COMMUTATING = 10;
-    WHEEL_1_ENCODER_NOT_COMMUTATING = 11;
-    WHEEL_2_ENCODER_NOT_COMMUTATING = 12;
-    WHEEL_3_ENCODER_NOT_COMMUTATING = 13;
-    WHEEL_0_HALL_SENSOR_STUCK_LOW   = 14;
-    WHEEL_1_HALL_SENSOR_STUCK_LOW   = 15;
-    WHEEL_2_HALL_SENSOR_STUCK_LOW   = 16;
-    WHEEL_3_HALL_SENSOR_STUCK_LOW   = 17;
-    DRIBBLER_HALL_SENSOR_STUCK_LOW  = 18;
-    WHEEL_0_HALL_SENSOR_STUCK_HIGH  = 19;
-    WHEEL_1_HALL_SENSOR_STUCK_HIGH  = 20;
-    WHEEL_2_HALL_SENSOR_STUCK_HIGH  = 21;
-    WHEEL_3_HALL_SENSOR_STUCK_HIGH  = 22;
-    DRIBBLER_HALL_SENSOR_STUCK_HIGH = 23;
-
-    // Edge triggered events
-    ICB_CRC_ERROR                        = 24;
-    RECEIVE_FRAME_CHECK_SEQUENCE_FAILURE = 25;
-    CRASHED_CORE_DUMPED                  = 26;
-    CRASHED_NO_CORE_DUMP                 = 27;
-
-    // Radio Dongle Errors
-    ROBOT_DEAD            = 28;
-    ESTOP_BROKEN          = 29;
-    RX_FCS_FAIL           = 30;
-    SECOND_DONGLE         = 31;
-    TRANSMIT_QUEUE_FULL   = 32;
-    RECEIVE_QUEUE_FULL    = 33;
-    PACKET_ABOVE_MAX_SIZE = 34;
-}
-
-/* Firmware build information  */
-message FirmwareStatus
-{
-    // The microcontroller firmware build ID.
-    uint32 fw_build_id = 1;
+    LOW_CAP            = 1;
+    LOW_BATTERY        = 2;
+    HIGH_BOARD_TEMP    = 3;
+    DRIBBLER_MOTOR_HOT = 4;
 }
 
-/* Encoder data from a single motor
- * Note that the rotation is from the perspective of rear of the motor, looking down
- * the shaft starting from the motor body
- */
 message DriveUnit
 {
-    float wheel_ang_vel_rad_s = 1;
-    float wheel_position_rad  = 2;
+    // The current RPM of the wheel
+    uint32 wheel_rpm = 1;
 }
 
 /* Data from all four drive units */
-message DriveUnitStatus
+message MotorStatus
```

Comment:
adjust comment to include dribblerstatus

Reply:
updated